crates/core/component/auction/src/auction/dutch.rs

```diff
@@ -0,0 +1,192 @@
+                    anyhow!("DutchAuctionDescription message is missing an output identifier")
+                })?
+                .try_into()?,
+            max_output: msg
+                .max_output
+                .ok_or_else(|| anyhow!("DutchAuctionDescription message is missing max output"))?
+                .try_into()?,
+            min_output: msg
+                .min_output
+                .ok_or_else(|| anyhow!("DutchAuctionDescription message is missing min output"))?
+                .try_into()?,
+            start_height: msg.start_height,
+            end_height: msg.end_height,
+            step_count: msg.step_count,
+            nonce: msg.nonce.as_slice().try_into()?,
+        };
+        Ok(d)
+    }
+}
+/* ********************************** */
+
+/// A stateful description of a dutch auction, recording its state (via a sequence number),
+/// the current position id associated to it (if any), and its amount IO.
+/// # State
+/// We record the state of the dutch auction via an untyped `u64` instead of an enum.
+/// This futureproof support for auction types that have a richer state machine e.g. allows
+/// claiming a withdrawn auction multiple times, burning and minting a new withdrawn auction
+/// with an incremented sequence number.
+///
+/// For Dutch auctions:
+///
+///   ┌───┐            ┌───┐             ┌───┐
+///   │ 0 │───Closed──▶│ 1 │──Withdrawn─▶│ 2 │
+///   └───┘            └───┘             └───┘
+///     ▲                                     
+///     │                                     
+///  Opened                                   
+///     │                                     
+///
+#[derive(Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(try_from = "pb::DutchAuctionState", into = "pb::DutchAuctionState")]
+pub struct DutchAuctionState {
+    pub sequence: u64,
+    pub current_position: Option<position::Id>,
+    pub next_trigger: u64,
+    pub input_reserves: Amount,
+    pub output_reserves: Amount,
+}
+
+/* Protobuf impls for `DutchAuctionState` */
+impl DomainType for DutchAuctionState {
+    type Proto = pb::DutchAuctionState;
+}
+
+impl From<DutchAuctionState> for pb::DutchAuctionState {
+    fn from(domain: DutchAuctionState) -> Self {
+        Self {
+            seq: domain.sequence,
+            current_position: domain.current_position.map(Into::into),
+            next_trigger: domain.next_trigger,
+            input_reserves: Some(domain.input_reserves.into()),
+            output_reserves: Some(domain.output_reserves.into()),
+        }
+    }
+}
+
+impl TryFrom<pb::DutchAuctionState> for DutchAuctionState {
+    type Error = anyhow::Error;
+
+    fn try_from(msg: pb::DutchAuctionState) -> Result<Self, Self::Error> {
+        let domain_position_id: Option<position::Id> =
+            if let Some(pb_position_id) = msg.current_position {
+                Some(pb_position_id.try_into()?)
+            } else {
+                None
+            };
```

Comment:
```suggestion
        let current_position =
            msg.current_position.map(TryInto::try_into).transpose()?;
```

..but i wouldn't block on this, just a subjective tweak.